staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -197,6 +197,7 @@ func (c *CloudNodeLifecycleController) MonitorNodes(ctx context.Context) {
 }
 
 // shutdownInCloudProvider returns true if the node is shutdown on the cloud provider
+// If provider id in spec is empty it calls instanceId with node name to get provider id
```

Comment:
shutdownInCloudProvider and ensureNodeExistsByProviderID seems to be used only once in the MonitorNodes, can we get the provider and pass it to the function instead of calling twice?

Reply:
This would be quite some refactor because the whole interface 
```
type InstancesV2 interface {
	InstanceExists(ctx context.Context, node *v1.Node) (bool, error)
	InstanceShutdown(ctx context.Context, node *v1.Node) (bool, error)
	InstanceMetadata(ctx context.Context, node *v1.Node) (*InstanceMetadata, error)
}
```
relies on passing the note object and not just an id. Asked via slack (https://kubernetes.slack.com/archives/C718BPBQ8/p1694687948972739) if that's desired here or not

Reply:
As discussed, changed only signatures for
```
func shutdownInCloudProvider(ctx context.Context, cloud cloudprovider.Interface, node *v1.Node, providerID string) (bool, error)
```
and
```
func ensureNodeExistsByProviderID(ctx context.Context, cloud cloudprovider.Interface, node *v1.Node, providerID string)
```

Reply:
And after final discussion, back to the original function signature